src/biogeophys/SoilTemperatureMod.F90

```diff
@@ -919,17 +944,51 @@ subroutine PhaseChangeH2osfc (bounds, num_nolakec, filter_nolakec, &
                ! remove ice from h2osfc
                h2osfc(c) = h2osfc(c) + xm(c)
 
-               xmf_h2osfc(c) = frac_h2osfc(c)*hm(c)
-
+!               xmf_h2osfc(c) = frac_h2osfc(c)*hm(c)
+!scs
+               xmf_h2osfc(c) = hm(c)
+!scs
                qflx_h2osfc_to_ice(c) = -xm(c)/dtime
 
+
                ! update snow depth
                if (frac_sno(c) > 0 .and. snl(c) < 0) then 
                   snow_depth(c)=h2osno(c)/(rho_avg*frac_sno(c))
                else
                   snow_depth(c)=h2osno(c)/denice
                endif
 
+! adjust temperature of lowest snow layer to account for addition of ice
+               if (snl(c) == 0) then
+                  !initialize for next time step
+                  t_soisno(c,0) = t_h2osfc(c)
+                  eflx_h2osfc_to_snow_col(c) = 0.
+               else if (snl(c) == -1) then
+                  c1=frac_sno(c)*(dtime/fact(c,0) - dhsdT(c)*dtime)
+                  if ( frac_h2osfc(c) /= 0.0_r8 )then
+!                     c2=frac_h2osfc(c)*(-cpliq*xm(c))
+                     c2=(-cpliq*xm(c))
+                  else
+                     c2=0.0_r8
+                  end if
+                  t_soisno(c,0) = (c1*t_soisno(c,0)+ c2*t_h2osfc(c)) &
+                       /(c1 + c2)             
+
+                  eflx_h2osfc_to_snow_col(c) =(t_h2osfc(c)-t_soisno(c,0))*c2/dtime
+               else
```

Comment:
It looks like this `else` block is exactly the same as the above `else if` block, except that here there is no `dhsdT` in the calculation of `c1`. To reduce this code duplication, can you do something like this?:

```Fortran
               if (snl(c) == 0) then
                  !initialize for next time step
                  t_soisno(c,0) = t_h2osfc(c)
                  eflx_h2osfc_to_snow_col(c) = 0.
               else
                  if (snl(c) == -1) then
                     somevar = dhsdT(c)
                  else
                     somevar = 0._r8
                  end if
                  c1=frac_sno(c)*(dtime/fact(c,0) - somevar*dtime)
                  ! Rest of the code here
```